### PR TITLE
[UT] add unit test for selfAttention, and a small fix

### PR DESCRIPTION
**Pull request by pujiang2018**

### CODE REVIEW

File: tests/ut/attention_kernels_test.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    bfloat16_t *qkv = (bfloat16_t *)xft::alloc(totalTokens * qkvStride * sizeof(bfloat16_t));
+
+    bfloat16_t *kCache = kvCache;
+    bfloat16_t *vCache = kvCache + batchSize * maxTokens * (kvHeadNum * headSize);
+
+    bfloat16_t *query = qkv;
+    bfloat16_t *key = qkv + headSize * qHeadNum;
+    bfloat16_t *value = qkv + headSize * (qHeadNum + kvHeadNum);
+
+    for (int i = 0; i < totalTokens * qkvStride; i++) {
```
Comment (changqi1) [inline tests/ut/attention_kernels_test.cpp:137]: #pragma omp parallel for

**Comment by changqi1** [comment]
When token_length < 1K use fusedAttention, >1K use FlashAttention. Do we need to test FlashAttention?

File: tests/ut/attention_kernels_test.cpp (hunk 1 above)
Comment (pujiang2018) [reply tests/ut/attention_kernels_test.cpp:137]: when use rand(), openmp will make it slower. Here is an article:
https://mahesh-hegde.github.io/posts/omp_rand_critical_section/

**Comment by pujiang2018** [comment]
> When token_length < 1K use fusedAttention, >1K use FlashAttention. Do we need to test FlashAttention?

Need, but in future.

**Review by changqi1 — APPROVED** [review]

**Pull request merged**